doc/Preferences.md

```diff
@@ -0,0 +1,179 @@
+Theia has a preference service which allows modules to get preference values, contribute default preferences and listen for preference changes.
+
+Preferences can be saved in the root of the workspace under `.theia/prefs.json` or under `$HOME/.theia/prefs.json`
```

Comment:
Good question! I think that because we're using os.homedir() from node 4+, it uses %USERPROFILE% variable on Windows, but yes I should have something about that in the document.